### PR TITLE
configurable subnet in EC2

### PR DESCRIPTION
**Pull request by jgleissner**
### What does this PR do? Why are we making this change?

This allows the user to set a subnet in the EC2 account configuration that will be used for uploading and testing instead of a temporary one. This can be used to avoid hitting a VPC count limit (issue  #482).

### How will these changes be tested?

Changes have been tested manually.

### How will this change be deployed? Any special considerations?

For supporting this feature properly, mash-client requires support for configuring the subnet. This is not implement yet.

### CODE REVIEW

File: mash/services/testing/ec2_job.py [hunk 1]
```diff
@@ -104,6 +104,10 @@ def run_job(self):
                 'ssh_user': self.ssh_user,
                 'tests': self.tests
             }
+            subnet_id = info.get('subnet')
+            if subnet_id:
+                img_proof_kwargs['subnet_id'] = subnet_id
+                img_proof_kwargs['security_group_id'] = info.get('security_groups')
```
Comment (smarlowucf) [inline mash/services/testing/ec2_job.py]: This isn't needed unless security_groups option is fully implemented.

File: mash/services/testing/img_proof_helper.py [hunk 2]
```diff
@@ -37,15 +38,14 @@
 def img_proof_test(
     results, cloud=None, access_key_id=None, description=None, distro=None,
     image_id=None, instance_type=None, img_proof_timeout=None, region=None,
-    secret_access_key=None, service_account_credentials=None,
-    ssh_private_key_file=None, ssh_user=None, tests=None, fallback_regions=None
+    secret_access_key=None, security_group_id=None,
```
Comment (smarlowucf) [inline mash/services/testing/img_proof_helper.py]: No need for security_group_id option yet.

File: mash/services/uploader/ec2_job.py [hunk 3]
```diff
@@ -136,21 +137,31 @@ def run_job(self):
...
 
-                self.ec2_upload_parameters['vpc_subnet_id'] = vpc_subnet_id
-                self.ec2_upload_parameters['security_group_ids'] = \
-                    security_group_id
+                subnet = info.get('subnet')
+                if subnet:
+                    vpc_id = get_vpc_id_from_subnet(ec2_client, subnet)
+                    security_group_id = ec2_setup.create_security_group(vpc_id=vpc_id)
+
+                    self.ec2_upload_parameters['vpc_subnet_id'] = subnet
```
Comment (smarlowucf) [inline mash/services/uploader/ec2_job.py]: Can be removed...

File: mash/services/uploader/ec2_job.py [hunk 4]
```diff
@@ -136,21 +137,31 @@ def run_job(self):
...
+                    self.ec2_upload_parameters['security_group_ids'] = \
+                        security_group_id
+                else:
+                    vpc_subnet_id = ec2_setup.create_vpc_subnet()
+                    security_group_id = ec2_setup.create_security_group()
+
+                    self.ec2_upload_parameters['vpc_subnet_id'] = vpc_subnet_id
```
Comment (smarlowucf) [inline mash/services/uploader/ec2_job.py]: Can be moved out of conditional to remove duplication.

File: mash/services/uploader/ec2_job.py [hunk 5]
```diff
@@ -136,21 +137,31 @@ def run_job(self):
...
+                subnet = info.get('subnet')
+                if subnet:
+                    vpc_id = get_vpc_id_from_subnet(ec2_client, subnet)
+                    security_group_id = ec2_setup.create_security_group(vpc_id=vpc_id)
+
+                    self.ec2_upload_parameters['vpc_subnet_id'] = subnet
+                    self.ec2_upload_parameters['security_group_ids'] = \
+                        security_group_id
+                else:
+                    vpc_subnet_id = ec2_setup.create_vpc_subnet()
```
Comment (smarlowucf) [inline mash/services/uploader/ec2_job.py]: Can be renamed to subnet...

File: mash/services/uploader/ec2_job.py (hunk 4 above)
Comment (smarlowucf) [reply mash/services/uploader/ec2_job.py]: E.g.

```
subnet = info.get('subnet')
if subnet:
    vpc_id = get_vpc_id_from_subnet(ec2_client, subnet)
    security_group_id = ec2_setup.create_security_group(vpc_id=vpc_id)
else:
    subnet = ec2_setup.create_vpc_subnet()
    security_group_id = ec2_setup.create_security_group()

self.ec2_upload_parameters['vpc_subnet_id'] = subnet
self.ec2_upload_parameters['security_group_ids'] = security_group_id
```

File: mash/services/testing/ec2_job.py (hunk 1 above)
Comment (jgleissner) [reply mash/services/testing/ec2_job.py]: That's a leftover, sorry. I had initially added a `security_group_id` parameter and later removed it, but evidently not completely.

File: mash/services/testing/img_proof_helper.py (hunk 2 above)
Comment (jgleissner) [reply mash/services/testing/img_proof_helper.py]: Same here.

File: mash/services/uploader/ec2_job.py (hunk 4 above)
Comment (jgleissner) [reply mash/services/uploader/ec2_job.py]: Ack.

**Pull request merged**